packages/utilities/src/utils/useId.ts

```diff
@@ -5,7 +5,15 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+import { useId as useReachId } from '@reach/auto-id';
+
+let idCounter = 0;
+
 /**
- * Autogenerate IDs to facilitate WAI-ARIA and server rendering from @reach, https://reach.tech/auto-id.
+ * Hook for generating a unique ID
+ *
+ * @param id A user provided ID
+ *
+ * @returns A generated ID that can be passed to accessibility attributes
  */
-export * from '@reach/auto-id';
+export const useId = (id: any) => useReachId(id) || `ssr:${idCounter++}`;
```

Comment:
I could be wrong, but won't the ID change between server and client in this strategy?

Reply:
Ah, I see now, even [`accessible-ui/use-id`](https://github.com/accessible-ui/use-id/blob/master/src/index.tsx) has a similar strategy. This works for me. :)